plugins/SitesManager/API.php

```diff
@@ -360,12 +360,18 @@ public function getAllSitesId()
      * @param bool $fetchAliasUrls
      * @param false|string $pattern
      * @param false|int    $limit
+     * @param []|int[] $sitesToExclude optional array of Integer IDs of sites to exclude from the result.
      * @return array for each site, an array of information (idsite, name, main_url, etc.)
      */
-    public function getSitesWithAdminAccess($fetchAliasUrls = false, $pattern = false, $limit = false)
+    public function getSitesWithAdminAccess($fetchAliasUrls = false, $pattern = false, $limit = false, $sitesToExclude = [])
     {
         $sitesId = $this->getSitesIdWithAdminAccess();
 
+        // Remove the sites to exclude from the list of IDs.
+        if (is_array($sitesId) && is_array($sitesToExclude) && count($sitesToExclude)) {
```

Comment:
just a suggestion could just check if `!empty($sitesToExclude)`, I think `is_array([])` will always return turn
```suggestion
        if (!empty($sitesToExclude)) {
```

Reply:
@peterhashair Thank you for the suggestion. I chose to use `is_array` because this PHP code isn't strongly typed and I wanted to make sure that `$sitesToExclude` is an array and not something else. Using `empty` could return a false positive if `$sitesToExclude` were a string or something else. I wanted to make sure that I was dealing with arrays before calling `array_diff` which expects arrays. Yes `is_array([])` returns true. Checking the `count` of `$sitesToExclude` prevents checking an empty array. I guess I could have checked the count of `$sitesId` too, but didn't feel it was necessary.

Reply:
@snake14 that makes sense.